thavalon-server/src/game/state/on_mission.rs

```diff
@@ -133,7 +133,8 @@ impl OnMission {
 
 impl GameState<WaitingForAgravaine> {
     pub fn handle_declaration(mut self, player: &str) -> ActionResult {
-        let mission_number = self.mission();
+        // Subtract one since the mission Agravaine failing is the one prior to the current propposal.
+        let mission_number = self.mission() - 1;
```

Comment:
meta-comment: right now, the backend infers the mission number and force counter based on other state. Would it be more reliable to introduce explicit fields for them?

Reply:
That's up to you. I think it's fine to keep it inferring for now, but I could go either way.